### PR TITLE
Feature/37901 register input question data

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
Question data should be passed from component to service instead of having the service reading the DOM to acquire them.

### CODE REVIEW

File: pupil-spa/src/app/services/register-input/registerInput.service.spec.ts [hunk 1]
```diff
@@ -70,8 +80,13 @@ describe('RegisterInputService', () => {
...
         type: 'mousedown', which: 1, currentTarget: null
       };
-      registerInputService.addEntry(event);
-      registerInputService.addEntry(event);
+      const questionData = {
+        questionNumber: '1',
+        factor1: '1',
+        factor2: '12',
+      };
+      registerInputService.addEntry(event, questionData);
```
Comment (GuyHarwood) [inline pupil-spa/src/app/services/register-input/registerInput.service.spec.ts]: are 2 subsequent calls intentional?

Comment (GeorgeChatzigiannis) [reply pupil-spa/src/app/services/register-input/registerInput.service.spec.ts]: I didn't find any particular value or reason behind having a second call so I removed it.

Comment (GeorgeChatzigiannis) [reply pupil-spa/src/app/services/register-input/registerInput.service.spec.ts]: thank you for spotting it

**Comment by GuyHarwood** [comment]
@jon-shipley as long as azure does not use `yarn start` they should be ok.

**Comment by GeorgeChatzigiannis** [comment]
`prestart` has been added back

**Pull request merged**